File: interviewbit-strings-length-of-last-word.java
public class Solution {
    public int lengthOfLastWord(final String a) {
        String trimmed = a.trim();
        for (int i = trimmed.length() - 1; i >= 0; i--) {
            if (trimmed.charAt(i) == ' ') {
                return trimmed.length() - i - 1;
            }
        }
        return trimmed.length();
    }
}

File: interviewbit-strings-length-of-last-word.py
class Solution:
    # @param A : string
    # @return an integer
    def lengthOfLastWord(self, A):
        last_char = len(A) - 1
        while last_char >= 0 and A[last_char] == " ":
            last_char -= 1

        for j in range(last_char, -1, -1):
            if A[j] == " ":
                return last_char - j

        return last_char + 1
